provisioning_client/tests/common_prov_e2e/prov_hsm/x509_msr.c

```diff
@@ -161,20 +159,21 @@ static int generate_root_ca_info(X509_CERT_INFO* x509_info, RIOT_ECC_SIGNATURE*
             result = 0;
         }
     }
+    ecc_signature_destroy(&tbs_sig);
     return result;
 }
 
-static int produce_device_cert(X509_CERT_INFO* x509_info, RIOT_ECC_SIGNATURE tbs_sig)
+static int produce_device_cert(X509_CERT_INFO* x509_info)
 {
     int result;
     uint8_t der_buffer[DER_MAX_TBS] = { 0 };
     DERBuilderContext der_ctx = { 0 };
     RIOT_STATUS status;
-
+    RIOT_ECC_SIGNATURE tbs_sig = { 0 };
 
     // Build the TBS (to be signed) region of DeviceID Certificate
     DERInitContext(&der_ctx, der_buffer, DER_MAX_TBS);
-    if (X509GetDeviceCertTBS(&der_ctx, &X509_DEVICE_TBS_DATA, &x509_info->device_id_pub) != 0)
+    if (X509GetDeviceCertTBS(&der_ctx, &X509_DEVICE_TBS_DATA, &x509_info->device_id_pub, NULL, 0) != 0)
```

Comment:
I assume there is no root CA key available here?  Otherwise it would be good to include it on this call.

Reply:
It should be available by the time we hit this call, switched back to using CA key - Fixed